### DISCUSSION TITLE
Does array in REGO support the operation like append and extend

### DISCUSSION THREAD

**Discussion by lemonshir** [open-policy-agent/community#77] [OPA and Rego] [answered]
I'd like to know if operation like `append` and `extend` is supported by `array`. If yes, could you point me to any doc link? If not, what's the best practice to achieve this in the current REGO? Can it be added in REGO natively in the future?

**Comment by anderseknert** [comment, marked as answer]
Hi @lemonshir! 

Rego variables and values are immutable, so there's no way to modify them directly. Like most languages with immutable values and vars, you can however use functions that return _new_ values or collections to achieve the same outcome. For your example of appending to an array, you can use the `array.concat` built-in:

```rego
a := [1, 2, 3]
b := [4, 5, 6]
c := array.concat(a, b)

# c == [1, 2, 3, 4, 5, 6]
```
